src/octree/mod.rs

```diff
@@ -115,6 +115,47 @@ impl<'a> InternalIterator for PointsInBoxIterator<'a> {
     }
 }
 
+pub struct PointsInFrustumIterator<'a> {
+    octree_meta: &'a OctreeMeta,
+    frustum_matrix: &'a Matrix4<f32>,
+    intersecting_nodes: Vec<NodeId>,
+}
+
+impl<'a> InternalIterator for PointsInFrustumIterator<'a> {
+    fn size_hint(&self) -> Option<usize> {
+        None
+    }
+
+    fn for_each<F: FnMut(&Point)>(self, mut f: F) {
+        for node_id in &self.intersecting_nodes {
+            let iterator = NodeIterator::from_disk(&self.octree_meta, node_id)
+                .expect("Could not read node points");
+            iterator.for_each(|p| {
+                if !contains(self.frustum_matrix, &Point3::from_vec(p.position)) {
+                    return;
+                }
+                f(p);
+            });
+        }
+    }
+}
+
+// TODO(ksavinash9) confirm with @Sirver and @thomasschiwietz about clipping vector approach
+// https://stackoverflow.com/questions/6301085/how-to-check-if-an-object-lies-outside-the-clipping-volume-in-opengl
+//
+// TODO(ksavinash9) check with @Sirver if contains should be a trait for Matrix4
```

Comment:
I'd say this is fine, no need for a trait.

Reply:
Thanks